### ISSUE TITLE
Thread-local rand

### ISSUE THREAD

**Issue by Paril** [jdolan/quetoo#341]
By making a few small changes to Random(), it will parallelize much better. I didn't push this since I didn't want to regress any, but wanted to at least remember to do this at some point.

```
/**
 * @brief Returns a pseudo-random positive integer.
 *
 * Uses a Linear Congruence Generator, values kindly borrowed from glibc.
 * Look up the rules required for the constants before just replacing them;
 * performance is dictated by the selection.
 */
int32_t Random(void) {

	static uint32_t seed;
	static __thread uint32_t state = 0;
	static __thread _Bool uninitalized = true;

	if (uninitalized) {
		seed += (uint32_t) time(NULL);
		state = seed;
		uninitalized = false;
	}

	state = (1103515245 * state + 12345);
	return state & 0x7fffffff;
}
```

**Comment by jdolan** [comment]
Cool. As long as things still seem randomized, I'm fine with this.

**Comment by jdolan** [comment]
What are you waiting for? :)

**Comment by Paril** [comment]
Implemented 86a6e706f7d875510cb1124129cc2972c5f13109
